### ISSUE TITLE
NewsChannel is not returned by Message.MentionType.CHANNEL

### ISSUE THREAD

**Issue by RedcodesDev** [DV8FromTheWorld/JDA#2034]
[download]: https://bintray.com/dv8fromtheworld/maven/JDA/_latestVersion
[guild]: https://discord.gg/0hMr4ce0tIk3pSjp
[stack overflow]: https://stackoverflow.com/questions/tagged/java

## General Troubleshooting

  
- [X] I have checked for similar issues.
- [X] I have updated to the [latest JDA version][download].
- [X] I have checked the branches or the maintainers' PRs for upcoming bug fixes.

## Bug Report

Message.MentionType.CHANNEL does not return NewsChannels. This was discovered by @ItzKiwiSap in #2027.
I am using 5.0.0-alpha.6. I also tried v4 and it worked perfectly fine.

### Expected Behavior

Message.MentionType.CHANNEL should also return NewsChannels.

### Code Example or Reproduction Steps

```java
public void onMessageReceived(MessageReceivedEvent e) {

	List<IMentionable> mentionedChannels = e.getMessage().getMentions(Message.MentionType.CHANNEL);
	if (mentionedChannels.isEmpty()) {
		System.out.println("empty");
		return;
	}

}
```

1. Mention a normal TextChannel in a message - nothing will be returned.
2. Mention a NewsChannel in a message - empty will be returned, indicating that there is no Channel Mention in the Message according to the JDA.

### Code for JDABuilder or DefaultShardManagerBuilder Used

```java
JDABuilder builder = JDABuilder.createDefault(token);

builder.setActivity(Activity.watching("Bot starting..."));
builder.setStatus(OnlineStatus.IDLE);

List<GatewayIntent> intents = new ArrayList<GatewayIntent>();
intents.addAll(EnumSet.allOf(GatewayIntent.class));
intents.remove(GatewayIntent.GUILD_PRESENCES);

builder.setEnabledIntents(intents);
builder.setMemberCachePolicy(MemberCachePolicy.ALL);
		
builder.addEventListeners(new MessageTest());

```
### Exception or Error

N/A

**Comment by MinnDevelopment** [comment]
This will should get fixed in #2015

**Comment by MinnDevelopment** [comment]
You can now use [Message#getMentions](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/Message.html#getMentions()) and [Mentions#getChannels](https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/Mentions.html#getChannels()) as of 5.0.0-alpha.12
